app/connectors/MessageConnector.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import java.time.OffsetDateTime
+import java.util.UUID
+
+import com.google.inject.Inject
+import config.AppConfig
+import connectors.MessageConnector.EisSubmissionResult
+import connectors.MessageConnector.EisSubmissionResult._
+import models.{ItemId, MessageSender, MessageType, MovementMessageWithStatus, TransitWrapper}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.http.logging.Authorization
+import uk.gov.hmrc.http.logging.SessionId
+import uk.gov.hmrc.http.HttpClient
+import utils.Format
+import uk.gov.hmrc.http.HttpReads.Implicits.readRaw
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+class MessageConnector @Inject()(config: AppConfig, http: HttpClient)(implicit ec: ExecutionContext) {
+
+  def post(itemId: ItemId, message: MovementMessageWithStatus, dateTime: OffsetDateTime)(
+    implicit headerCarrier: HeaderCarrier
+  ): Future[EisSubmissionResult] = {
+
+    val xmlMessage = TransitWrapper(message.message).toString
+
+    val url = config.eisUrl
+
+    lazy val messageSender = MessageSender(itemId, message.messageCorrelationId)
+
+    val newHeaders = headerCarrier
+      .copy(authorization = Some(Authorization(s"Bearer ${config.eisBearerToken}")))
+      .withExtraHeaders(addHeaders(message.messageType, dateTime, messageSender): _*)
+
+    http
+      .POSTString[HttpResponse](url, xmlMessage)(readRaw, hc = newHeaders, implicitly)
+      .map {
+        case x if x.status == EisSubmissionSuccessful.httpStatus       => EisSubmissionSuccessful
+        case x if x.status == ErrorInPayload.httpStatus                => ErrorInPayload
+        case x if x.status == VirusFoundOrInvalidToken.httpStatus      => VirusFoundOrInvalidToken
+        case x if x.status == DownstreamInternalServerError.httpStatus => DownstreamInternalServerError
+        case x                                                         => UnexpectedHttpResponse(x)
+      }
+  }
```

Comment:
Maybe here we can extract mapping statuses to responses in a separate map and that inside map function only iterate over x.status? I mean sth like that:
`val statusesToHttpResponses = Map(EisSubmissionSuccessful.httpStatus -> EisSubmissionSuccessful, ErrorInPayload.httpStatus -> ErrorInPayload,...)`
and then: 
```
   http
      .POSTString[HttpResponse](url, xmlMessage)(readRaw, hc = newHeaders, implicitly)
      .map { x => statusesToHttpResponses.getOrElse(x.status, UnexpectedHttpResponse(x)) }
  }
```